### DISCUSSION TITLE
How Can I apply Community Toolkit StatusBarBehaviour from Style globally to all my content pages.

### DISCUSSION THREAD

**Discussion by himanshudwd765** [CommunityToolkit/Maui#997] [General]
I want to apply Community Toolkit [StatusBarBehaviour](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/behaviors/statusbar-behavior?tabs=android) to change status bar color from style to globally apply to all my content pages. I tried like this in my Resources/Styles.xaml file:

```
<ResourceDictionary 
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">

    <Style TargetType="ContentPage" x:Key="statusBarColor">
        <Setter Property="Behaviors">
            <Setter.Value>
                <toolkit:StatusBarBehavior StatusBarColor="Red"/>
            </Setter.Value>
        </Setter>
    </Style>
</ResourceDictionary>
```

The above code didn't update the StatusBar color.

**Comment by acaliaro** [comment]
I think you can use it in the first page of your application and all the pages have the same color.

for Example I have this in the first page

    <ContentPage.Behaviors>
        <mct:StatusBarBehavior StatusBarColor="{x:StaticResource Primary}" StatusBarStyle="LightContent" />
    </ContentPage.Behaviors>

**Reply by himanshudwd765** [reply]
This works 👍

**Comment by VladislavAntonyuk** [comment]
Remove x:Key="statusBarColor"

**Reply by himanshudwd765** [reply]
Removing x:key doesn't make any difference, no change in status bar color

**Reply by VladislavAntonyuk** [reply]
the full solution:
```
<ResourceDictionary 
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">

    <Style TargetType="ContentPage">
        <Setter Property="Behaviors">
            <Setter.Value>
               <toolkit:StatusBarBehavior StatusBarColor="{x:StaticResource Primary}" StatusBarStyle="LightContent" />
            </Setter.Value>
        </Setter>
    </Style>
</ResourceDictionary>
```

**Reply by himanshudwd765** [reply]
Tried no change :( 

<img width="1433" alt="Screenshot 2023-02-15 at 2 40 51 PM" src="https://user-images.githubusercontent.com/16178794/218983752-57450d34-ebcf-4f0a-8ebc-76f6138d6bef.png">

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
